morphia/src/main/java/org/mongodb/morphia/dao/BasicDAO.java

```diff
@@ -57,7 +60,7 @@ public BasicDAO(final Class<T> entityClass, final MongoClient mongoClient, final
      * @param ds          the Datastore which gives access to the MongoDB instance for this DAO
      */
     public BasicDAO(final Class<T> entityClass, final Datastore ds) {
-        this.ds = (DatastoreImpl) ds;
+        this.ds = (org.mongodb.morphia.DatastoreImpl) ds;
```

Comment:
you can `@SuppressWarning` for *uses* of deprecated items but the compiler will still complain about imports.  Moving those uses to fully qualified class names circumvents that bug.